src/sql/workbench/parts/notebook/test/browser/notebookEditorModel.test.ts

```diff
@@ -691,6 +691,170 @@ suite('Notebook Editor Model', function (): void {
 		should(notebookEditorModel.lastEditFullReplacement).equal(false);
 	});
 
+	test('should not replace entire text model for content change with double quotes', async function (): Promise<void> {
+		await createNewNotebookModel();
+		let notebookEditorModel = await createTextEditorModel(this);
+		notebookEditorModel.replaceEntireTextEditorModel(notebookModel, undefined);
+
+		let newCell = notebookModel.addCell(CellTypes.Code);
+		setupTextEditorModelWithEmptyOutputs(notebookEditorModel, newCell);
+
+		addTextToBeginningOfTextEditorModel(notebookEditorModel, newCell, '"This text is in quotes"');
+		should(notebookEditorModel.editorModel.textEditorModel.getLineContent(9)).equal('                "\\"This text is in quotes\\""');
+
+		ensureStaticContentInOneLineCellIsCorrect(notebookEditorModel, newCell);
+		should(notebookEditorModel.lastEditFullReplacement).equal(false);
```

Comment:
@aaomidi - Didn't you recently discover something where async functions need to actually return a value after they're done or they'll cause issues with the test?

Reply:
I don't think so - it was unclear from the documentation but I dont think it has any impact.